packages/store/tests/ensure-store.spec.ts

```diff
@@ -0,0 +1,214 @@
+    expect(getSelectorMetadata(MyState)).toBeUndefined();
+  });
+
+  describe('Ensure store for plugins', () => {
+    @State({
+      name: 'myCounter',
+      defaults: 1
+    })
+    class MyCounterState {
+      @Selector()
+      @SelectorOptions({ suppressErrors: false })
+      public static canInheritSelectFn(state: number): number {
+        return state * 2;
+      }
+
+      @Action({ type: 'decrement' })
+      public decrement(): void {}
+    }
+
+    @State({
+      name: 'count',
+      defaults: 0,
+      children: [MyCounterState]
+    })
+    class CountState extends MyCounterState {
+      @Selector()
+      public static selectFn(state: number): number {
+        return state;
+      }
+
+      @Action({ type: 'increment' })
+      public addOne(): void {}
+
+      @Action({ type: 'increment' })
+      public addTwo(): void {}
+    }
+
+    beforeAll(() => {
+      TestBed.configureTestingModule({
+        imports: [NgxsModule.forRoot([CountState, MyCounterState])]
+      });
+    });
+
+    it('should get the meta data from the CountState', () => {
+      expect(getStoreMetadata(CountState)).toEqual({
+        name: 'count',
+        actions: {
+          increment: [
+            { fn: 'addOne', options: {}, type: 'increment' },
+            { fn: 'addTwo', options: {}, type: 'increment' }
+          ],
+          decrement: [{ fn: 'decrement', options: {}, type: 'decrement' }]
+        },
+        defaults: 0,
+        path: null,
+        instance: null,
+        selectFromAppState: null,
+        children: [MyCounterState]
+      });
+    });
+
+    it('should get the meta data from the MyCounterState', () => {
+      expect(getStoreMetadata(MyCounterState)).toEqual({
+        name: 'myCounter',
+        actions: { decrement: [{ fn: 'decrement', options: {}, type: 'decrement' }] },
+        defaults: 1,
+        path: null,
+        selectFromAppState: null,
+        children: undefined,
+        instance: null
+      });
+    });
+
+    it('should get the selector meta data from the CountState, MyCounterState', () => {
+      expect(getSelectorMetadata(CountState)).toBeUndefined();
+      expect(getSelectorMetadata(MyCounterState)).toBeUndefined();
+    });
+
+    it('should get the selector meta data from the CountState.selectFn', () => {
+      const metadata: SelectorMetaDataModel = getSelectorMetadata(CountState.selectFn);
+
+      expect(metadata.selectorName).toEqual('selectFn');
+      expect(metadata.containerClass).toEqual(CountState);
+
+      expect(metadata.originalFn).not.toEqual(CountState.selectFn);
+      expect(extractFunctionBody(metadata.originalFn)).toEqual(
+        'function (state) { return state; }'
+      );
+
+      expect(metadata.originalFn!(1)).toEqual(1); // state => state
+
+      expect(extractFunctionBody(metadata.getSelectorOptions)).toEqual(
+        'function () { return getCustomSelectorOptions(selectorMetaDataClone, getExplicitSelectorOptions()); }'
```

Comment:
Isn't it dangerous to compare with the string here? I mean if we change any functionality of the `getSelectorOptions` in the future we will need to always fix this test.

Reply:
done